control-plane/api-gateway/binding/validation.go

```diff
@@ -167,43 +206,92 @@ func (m mergedListeners) validateHostname(index int, listener gwv1beta1.Listener
 // validateTLS validates that the TLS configuration for a given listener is valid and that
 // the certificates that it references exist.
 func validateTLS(gateway gwv1beta1.Gateway, tls *gwv1beta1.GatewayTLSConfig, resources *common.ResourceMap) (error, error) {
-	namespace := gateway.Namespace
-
+	// If there's no TLS, there's nothing to validate
 	if tls == nil {
 		return nil, nil
 	}
 
-	var err error
+	// Validate the certificate references and then return any error
+	// alongside any TLS configuration error that we find below.
+	refsErr := validateCertificateRefs(gateway, tls.CertificateRefs, resources)
+
+	if tls.Mode != nil && *tls.Mode == gwv1beta1.TLSModePassthrough {
+		return errListenerNoTLSPassthrough, refsErr
+	}
+
+	if err := validateTLSOptions(tls.Options); err != nil {
+		return err, refsErr
+	}
+
+	return nil, refsErr
+}
 
-	for _, cert := range tls.CertificateRefs {
-		// break on the first error
+func validateCertificateRefs(gateway gwv1beta1.Gateway, refs []gwv1beta1.SecretObjectReference, resources *common.ResourceMap) error {
```

Comment:
The diff here is easiest to view in split mode IMO. It amounts to splitting the existing logic out into some more granular functions and adding a new function to validate the TLS options

<img src="https://github.com/hashicorp/consul-k8s/assets/3476400/5b00d5ec-475b-4ec7-9b92-ef96ab3a696f" width="250">